soc/arm64/rockchip/rk3399/mmu_regions.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 Huawei France Technologies SASU
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/devicetree.h>
+#include <soc.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/arch/arm64/arm_mmu.h>
+#include <zephyr/arch/arm64/kaddr.h>
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 0),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_DEFAULT_SECURE_STATE),
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 1),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_DEFAULT_SECURE_STATE),
+
+	MMU_REGION_ENTRY("UART",
+			      DT_REG_ADDR(DT_INST(0, rockchip_rk3399_uart)),
+			      KADDR_FROM_PADDR_CT(DT_REG_ADDR(DT_INST(0, rockchip_rk3399_uart))),
+			      DT_REG_SIZE(DT_INST(0, rockchip_rk3399_uart)),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_DEFAULT_SECURE_STATE),
```

Comment:
do not map this. Just modify the UART driver to support the device MMIO API (see `include/zephyr/sys/device_mmio.h`).